packages/app-mobile/components/TagEditor.tsx

```diff
@@ -0,0 +1,221 @@
+import * as React from 'react';
+
+import { StyleSheet, View, Text, ScrollView, ViewStyle } from 'react-native';
+import { _ } from '@joplin/lib/locale';
+import { themeStyle } from './global-style';
+import ComboBox, { Option } from './ComboBox';
+import IconButton from './IconButton';
+import { useCallback, useMemo } from 'react';
+import { TagEntity } from '@joplin/lib/services/database/types';
+import { Divider } from 'react-native-paper';
+
+export enum TagEditorMode {
+	Large,
+	Compact,
```

Comment:
The tag editor dialog currently uses the `Large` mode. The `Compact` mode makes the tag editor take up less space and is used in the document scanner screen (to be added in a separate pull request).